Sources/Machine/rbm_spin.cc

```diff
@@ -46,20 +46,40 @@ RbmSpin::RbmSpin(std::shared_ptr<const AbstractHilbert> hilbert, Index nhidden,
                      << nhidden << " != " << alpha << " * " << nvisible);
   }
   nhidden = std::max(nhidden, alpha * nvisible);
-
-  W_.resize(nvisible, nhidden);
+  /* Initialization of the instance of RbmSpin class (defined in 
+   * header) Variables are:
+   * hilbert - the variable containing information about physical system
+   * nhidden - number of neurons in hidden layer defined by larger of the values
+   *           (alpha * number of neuron from visible layer) or nhidden
+   * alpha - density defined as (number of neurons in hidden layer / number of
+   *         neurons in visible layer)
+   * usea - if true use biases in visible layer a*output
+   * useb - if true use biases in hidden layer b*output
```

Comment:
I'd rather place those comments in the header than here, because it's where you look up information for class members

Reply:
Thanks @PhilipVinc for the quick review. I just made an update to it as per your suggestions.